### ISSUE TITLE
Restore pre-commit container image tag pinning

### ISSUE THREAD

**Issue by blackliner** [hadolint/hadolint#921]
- [x] This is a bug report
- [ ] This is a feature request
- [ ] I searched existing issues before opening this one

### Expected behavior
When updating the hadolint version in the pre-commit configuration, I expect the used hadolint container image to match that version.

### Actual behavior
Since https://github.com/hadolint/hadolint/pull/814, hadolint is using the latest tag, and not pulling the requested tag.

### Steps to reproduce the behavior
See https://github.com/hadolint/hadolint/pull/814 for how to configure pre-commit to a specific release and how to check the currently used container image.

### Reason
https://github.com/hadolint/hadolint/blob/master/.pre-commit-hooks.yaml#L7 is not pinning the docker version tag anymore. This happened in https://github.com/hadolint/hadolint/pull/814, which needs to be reverted and fixed in the release process.
Using `latest` is considered bad practice:
- https://vsupalov.com/docker-latest-tag/
- https://www.howtogeek.com/devops/understanding-dockers-latest-tag/
- https://medium.com/@mccode/the-misunderstood-docker-tag-latest-af3babfd6375

**Comment by lorenzo** [comment]
I would gladly accept a PR that updates the file after a new release

**Comment by blackliner** [comment]
After a release would be too late, it needs to be done as for the commit that is then tagged for the release, right?

**Comment by lorenzo** [comment]
I think whatever we put in the master branch is ok, so it should probably happen after the release

**Comment by blackliner** [comment]
No, because it will not take whatever is latest on branch (bad idea), but use the git tag from the release. So the tag would point to `COMMIT`, and the actual change of the version would be on `COMMIT+1`

**Comment by KSmanis** [comment]
@blackliner is correct, the Docker tag in `.pre-commit-hooks.yaml` should be bumped before creating the Git tag, i.e., it should be part of the tagged tree.

@lorenzo how are the Git tags created? Manually or is there a release script? I did a cursory search in the repo and didn't find one.

**Comment by lorenzo** [comment]
Currently they are created manually @KSmanis. I'd be happy to accept any contributions to automate this. For instance a manually dispatched GitHub action that tags a new version and does the rest

**Comment by KSmanis** [comment]
Judging from your last three tags, you just bump `hadolint.cabal` and `package.yaml` before git tagging. Am I missing any other steps?
